File: java/maxvalue/src/test/java/dev/jgn/SolutionTest.java
package dev.jgn;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class SolutionTest {

    @Test
    void TestCaseOne() {
        var numbers = new Integer[]{4, 7, 2, 8, 10, 9};
        var expected = 10;
        var result = Solution.maxValue(numbers);
        assertEquals(expected, result);
    }

    @Test
    void TestCaseTwo() {
        var numbers = new Integer[]{10, 5, 40, 42};
        var expected = 42;
        var result = Solution.maxValue(numbers);
        assertEquals(expected, result);
    }

    @Test
    void TestCaseThree() {
        var numbers = new Integer[]{-5, -2, -1, -11};
        var expected = -1;
        var result = Solution.maxValue(numbers);
        assertEquals(expected, result);
    }

    @Test
    void TestCaseFour() {
        var numbers = new Integer[]{42};
        var expected = 42;
        var result = Solution.maxValue(numbers);
        assertEquals(expected, result);
    }

    @Test
    void TestCaseFive() {
        var numbers = new Integer[]{1000, 8};
        var expected = 1000;
        var result = Solution.maxValue(numbers);
        assertEquals(expected, result);
    }

    @Test
    void TestCaseSix() {
        var numbers = new Integer[]{1000, 8, 9000};
        var expected = 9000;
        var result = Solution.maxValue(numbers);
        assertEquals(expected, result);
    }

    @Test
    void TestCaseSeven() {
       var numbers = new Integer[]{2, 5, 1, 1, 4};
       var expected = 5;
       var result = Solution.maxValue(numbers);
       assertEquals(expected, result);
    }

    @Test
    void TestCaseEight() {
        var numbers = new Integer[]{1, 1, 1, 1, 1, 1};
        var expected = 1;
        var result = Solution.maxValue(numbers);
        assertEquals(expected, result);
    }
}

File: go/treeIncludes/solution_test.go
package solution

import "testing"

func TestCaseOne(t *testing.T) {
	a := NewNode("a")
	b := NewNode("b")
	c := NewNode("c")
	d := NewNode("d")
	e := NewNode("e")
	f := NewNode("f")

	a.left = b
	a.right = c
	b.left = d
	b.right = e
	c.right = f

	got := treeIncludes(a, "e")
	want := true

	if got != want {
		t.Errorf("got %v, want %v", got, want)
	}
}

func TestCaseTwo(t *testing.T) {
	a := NewNode("a")
	b := NewNode("b")
	c := NewNode("c")
	d := NewNode("d")
	e := NewNode("e")
	f := NewNode("f")

	a.left = b
	a.right = c
	b.left = d
	b.right = e
	c.right = f

	got := treeIncludes(a, "z")
	want := false

	if got != want {
		t.Errorf("got %v, want %v", got, want)
	}
}
func TestCaseThree(t *testing.T) {
	a := NewNode("a")
	b := NewNode("b")
	c := NewNode("c")
	d := NewNode("d")
	e := NewNode("e")
	f := NewNode("f")

	a.left = b
	a.right = c
	b.left = d
	b.right = e
	c.right = f

	got := treeIncludes(a, "a")
	want := true

	if got != want {
		t.Errorf("got %v, want %v", got, want)
	}
}

func TestCaseFour(t *testing.T) {
	a := NewNode("a")
	b := NewNode("b")
	c := NewNode("c")
	d := NewNode("d")
	e := NewNode("e")
	f := NewNode("f")

	a.left = b
	a.right = c
	b.left = d
	b.right = e
	c.right = f

	got := treeIncludes(a, "j")
	want := false

	if got != want {
		t.Errorf("got %v, want %v", got, want)
	}
}

func TestCaseFive(t *testing.T) {
	got := treeIncludes(nil, "x")
	want := false

	if got != want {
		t.Errorf("got %v, want %v", got, want)
	}
}
